internal/types/compare.go

```diff
@@ -255,3 +254,130 @@ func compareNumbers(a float64, b int64) CompareResult {
 
 	return CompareResult(bigA.Cmp(bigB))
 }
+
+// dataTypeOrderResult represents the comparison order of data types.
+type dataTypeOrderResult uint8
```

Comment:
I wonder if we could come up with a better name…

@rumyantseva thoughts?

Reply:
Maybe one "pro" of dedicating a separate package is to decrease the cognitive load... Fewer lines in the file, simpler mental context... But still, it's rather a random idea.

Reply:
> Something like `typeInComparisonOrder` looks pretty much similar to how it's already 

@rumyantseva, how about `sequenceDatatype`?

Reply:
@AlekSi I believe, as the main guard of code standards in this repo, you should be the decision-maker here ;)